content/en/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions.md

```diff
@@ -1066,6 +1066,33 @@ message will be used instead.
 `messageExpression` is a CEL expression, so the restrictions listed in [Resource use by validation functions](#resource-use-by-validation-functions) apply. If evaluation halts due to resource constraints 
 during `messageExpression` execution, then no further validation rules will be executed.
 
+#### The reason and fieldPath fields
+
+Besides the `message` and `messageExpression` fields, which defines the string reported for a validation rule failure,
+we have also added two more fields under `validation`:
+- `reason` field which allows user to specify a machine-readable validation failure reason when a request fails this validation rule.
+- `fieldPath` field which specify the field path returned when the validation fails.
```

Comment:
We like [timeless](https://developers.google.com/style/timeless-documentation) documentation: please avoid “we have also added”.

```suggestion

Setting `messageExpression` is optional.

#### The `message` field {#field-message}

If you want to set a static message, you can supply `message` rather than `messageExpression`.
The value of `message` is used as an opaque error string if validation fails.

Setting `message` is optional.

#### The `reason` field {#field-reason}

You can add a machine-readable validation failure reason within a `validation`, to be returned
whenever a request fails this validation rule.

<!-- an example here would be nice, but is hard to add using GitHub suggestions -->

Setting `reason` is optional.

#### The `fieldPath` field {#field-field-path}

You can specify the path to the data that have failed to validate.
```
NB. Can we explain _why_ it's useful to specify a `fieldPath`?

Reply:
Thank you for the review! I have updated the doc to align with the suggested format.
For `fieldPath`, we wanna user to be able to specify the accurate path when err returned instead of always using the path which `self` scoped. It is very usefully especially when do cross fields validation and user wanna the returned err indicate the path the err occurred. For example, if user trying to say a field has to be a different value than another field, something like `self.foo.test.a != self.b`. The validation has to be scoped high enough to have access to both field `a` and `b` but we wanna the returned err path to be on `a`, in this case having a fieldPath `.foo.test.a` will be great.
Hope that would make sense to you. Thank you^^

Reply:
Edit: I have also added explain in the doc with the example. Thank you